src/simulation.cpp

```diff
@@ -730,7 +730,7 @@ void transport_event_based()
       simulation::collision_queue.size()});
 
     // Execute event with the longest queue
-    if (max == 0) {
+    if (max == 0 && remaining_work == 0) {
```

Comment:
This works! The only downside is that if all particles die early, the loop could have to spin for a bit to reach the refill interval. The cost would be totally negligible, but if we wanted to avoid it, one method would be to change the logic in line 747 as well to 

```C++
if ( (event_iteration % settings::event_queue_refill_interval == 0 && remaining_work > 0) || max == 0 )
```

I'm fine with it either way -- perhaps the more complex/confusing logic is not worth saving a few cycles on an edge case.

Reply:
I agree that the more complex logic isn't worth a free spin, so am resolving this as no change is required if we both agree the current setup is alright.